### PR TITLE
Prescriptions tests

### PR DESCRIPTION
**Pull request by alexose**
Adding test scaffold for prescriptions

### CODE REVIEW

File: test/rx/containers/RxRefillsApp.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import RxRefillsApp from '../../../src/js/rx/containers/RxRefillsApp';
+import reducer from '../../../src/js/rx/reducers';
+
+const store = createStore(reducer);
+
+describe('<RxRefillsApp>', () => {
+  it('Sanity check the component renders', () => {
+    const mockRoutes = [{ path: '/fake' }];
+    const tree = SkinDeep.shallowRender(<EduBenefitsApp store={store} location={{ pathname: '/blah' }} route={{ childRoutes: mockRoutes }}/>);
```
Comment (U-DON) [inline test/rx/containers/RxRefillsApp.spec.jsx]: `EduBenefitsApp` -> `RxRefillsApp`?

**Comment by alexose** [comment]
A few helpful notes about writing tests for `rx`!

There is lots of debate about whether or not unit testing on the frontend means "testing against the DOM".  In this author's opinion, the answer is "no", and anything involving the DOM is technically an integration (or end-to-end, or e2e) test.

That said, it's easy to find yourself in a scenario where you want to test an _event_, or whether or not a certain element is visible.  Resist!  Rather than testing the event, just test the event handler.  Rather than checking for visibility, just make sure it has the correct className.  Good unit tests are ones that enforce separation of concern.

Now, on to e2e testing:

Sometimes our e2e stack can fail in an annoyingly opaque way.  Fortunately, nightwatch has a couple of good windows into what's going on:

* On test failure, it will automatically take a screenshot and put it in `vets-website/logs/screenshots/rx/`.  You can at least see what screen selenium was looking at
* This isn't enabled by default, but you can pass console errors through to your command line:

```
    client
      .getLog(result => {
        console.log(result);
      })
```

**Comment by alexose** [comment]
There's still a bit more to do around creating a mock API.  I'll be following the example that `healthcare-app` uses.

**Comment by alexose** [comment]
This is ready for review.  Thanks @U-DON @webinista  !

File: test/rx/containers/RxRefillsApp.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { expect } from 'chai';
+import { createStore } from 'redux';
+
+import RxRefillsApp from '../../../src/js/rx/containers/RxRefillsApp';
+import reducer from '../../../src/js/rx/reducers';
+
+const store = createStore(reducer);
+
+describe('<RxRefillsApp>', () => {
+  it('Sanity check the component renders', () => {
```
Comment (U-DON) [inline test/rx/containers/RxRefillsApp.spec.jsx]: `it('should render')`?

File: test/util/rx-helpers.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        "station-number":"12",
+        "is-refillable":false,
+        "is-trackable":false
+      },
+      "links":{
+        "self":"http://localhost:3000/rx/v1/prescriptions/746575",
+        "tracking":"http://localhost:3000/rx/v1/prescriptions/746575/trackings"
+      }
+    }
+	]
```
Comment (U-DON) [inline test/util/rx-helpers.js]: Stray tab.

**Comment by U-DON** [comment]
LGTM aside from nits.

Mostly makes sense to me, but would like to know how to run these. `npm run test:unit` and `npm run test:e2e`? Do those run all tests from the other projects as well, or would running them within the `rx` subfolder run it for `rx` only?

Also not really sure what is happening within `initApplicationSubmitMock`.

File: test/util/rx-helpers.js (hunk 3 above)
Comment (alexose) [inline test/util/rx-helpers.js]: ... Okay at this point I think vim is trolling me.

Also, these files should be getting linted automatically, so I'll look into that.

Comment (alexose) [inline test/util/rx-helpers.js]: Serves me right-- I put it between an `/*eslint-disable*/` section.

**Comment by alexose** [comment]
Valid questions!

How to run them:  Right now, you run `npm run test` (or `npm run test:unit`, etc).  This will run all tests-- Including non `rx` related tests.

Given that this is sort of slow, you can also run them by calling the test runners directly:

`node_modules/mocha/bin/mocha --recursive 'test/rx/**/*.unit.spec.js?(x)'`
`script/run-nightwatch.sh test/rx/00-required.e2e.spec.js`

Running the e2e tests requires that you A) Have selenium installed and B) exit `webpack-dev-server` and run `npm run build`.  This is sort of inconvenient-- Hopefully we'll automate all of these in the near future.

**Comment by alexose** [comment]
`initApplicationSubmitMock` is a little crazy, but pretty neat when you break it down.

The test server allows you to set up any endpoint and response you need during the test process.  Basically you POST some data to `/mock` that tells it what endpoint you need and it immediately begins listening.

**Comment by U-DON** [comment]
So if you needed to set up multiple endpoints and responses, does that mean you would call `request` for each one?

Also, I think you mentioned you were adding some more tests, so holding off on merging unless you say otherwise.

**Comment by alexose** [comment]
Exactly-- check out the `hca` example if you're curious.

I'll do the next set of tests in a separate PR.  Go ahead and merge this if you 👍 it.

**Pull request merged**